cloudformation/cicd-resources.yml

```diff
@@ -0,0 +1,288 @@
+      Type: String
+      Value: CHANGEME
+
+  dockerUsernameParameter:
+    Type: AWS::SSM::Parameter
+    Properties:
+      Description: Parameter for the docker username
+      Name: /Firespring/CiCd/DockerUsername
+      Type: String
+      Value: dockerboxcicdmachine
+
+  dockerPasswordParameter:
+    Type: AWS::SSM::Parameter
+    Properties:
+      Description: Parameter for the docker password
+      Name: /Firespring/CiCd/DockerPassword
+      Type: String
+      Value: CHANGEME
+
+  cicdArtifactsBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Sub cicd-artifacts-bucket-${AWS::AccountId}-${AWS::Region}
+      AccessControl: Private
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              KMSMasterKeyID: !Ref key
+              SSEAlgorithm: 'aws:kms'
+      LifecycleConfiguration:
+        Rules:
+          - Status: Enabled
+            AbortIncompleteMultipartUpload:
+              DaysAfterInitiation: 7
+            NoncurrentVersionExpirationInDays: 30
+          - Status: Enabled
+            ExpirationInDays: 90
+            Transitions:
+              - StorageClass: STANDARD_IA
+                TransitionInDays: 30
+      LoggingConfiguration:
+        DestinationBucketName: !ImportValue AccessLoggingBucket
+        LogFilePrefix: !Sub cicd-artifacts-bucket-${AWS::AccountId}-${AWS::Region}
+      VersioningConfiguration:
+        Status: Enabled
+
+  # TODO: Encryption at rest - what permissions does lambda need?
+  dlqTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      DisplayName: Notifications for failed Lambda function events
+  #      KmsMasterKeyId: !GetAtt key.KeyId
+
+  # NOTE: this requires additional setup in the console after the create finishes (by a github repo owner)
+  # See https://us-east-1.console.aws.amazon.com/codesuite/settings/connections
+  codestarConnection:
+    Type: AWS::CodeStarConnections::Connection
+    Properties:
+      ConnectionName: !Sub GitHubConnection-${AWS::Region}
+      ProviderType: GitHub
+
+  pipelineStateChangeLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub /aws/events/codepipeline-state-change
+      RetentionInDays: 30
+
+  pipelineStateChangeLogGroupResourcePolicy:
+    Type: AWS::Logs::ResourcePolicy
+    Properties:
+      PolicyName: events-logs-resource-policy
+      PolicyDocument: !Sub
+        - >
+          {
+            "Statement": [
+              {
+                "Action": [
+                  "logs:CreateLogStream",
+                  "logs:PutLogEvents"
+                ],
+                "Effect": "Allow",
+                "Principal": {
+                  "Service": [
+                    "events.amazonaws.com",
+                    "delivery.logs.amazonaws.com"
+                  ]
+                },
+                "Resource": "arn:aws:logs:region:account:log-group:/aws/events/*:*",
```

Comment:
I believe region/account need to use the aws variables. I'm assuming this is wrong in the script you copied this from as well.

Reply:
The [fdp-devops cicd-resources](https://github.com/firespring/fdp-devops/blob/e7176d7977085771156ba48fa4ac037a9084a852/aws/cloudformation/fdp/cicd-resources.yml#L625) has this too, along with the engineering-devops also has this.